src/HexagonAlignLoads.cpp

```diff
@@ -0,0 +1,192 @@
+#include "HexagonAlignLoads.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Scope.h"
+#include "Bounds.h"
+#include "ModulusRemainder.h"
+#include "Simplify.h"
+namespace Halide {
+namespace Internal {
+using std::vector;
+
+class HexagonAlignLoads : public IRMutator {
+public:
+    HexagonAlignLoads(Target t) : target(t) {}
+private:
+    Target target;
+
+    /** Alignment info for Int(32) variables in scope. */
+    Scope<ModulusRemainder> alignment_info;
+    using IRMutator::visit;
+    ModulusRemainder get_alignment_info(Expr e) {
+        return modulus_remainder(e, alignment_info);
+    }
+    void visit(const Load *op) {
+        debug(4) << "HexagonAlignLoads: Working on " << (Expr) op << "..\n";
+        Expr index = mutate(op->index);
+        if (op->type.is_vector()) {
+            bool external = op->image.defined();
+            if (external) {
+                debug(4) << "HexagonAlignLoads: Not dealing with an external image\n";
+                debug(4) << (Expr) op << "\n";
+                expr = op;
+                return;
+            }
+            else {
+                const Ramp *ramp = index.as<Ramp>();
+                const IntImm *stride = ramp ? ramp->stride.as<IntImm>() : NULL;
+                // We will work only on natural vectors supported by the target.
+                if (ramp->lanes != target.natural_vector_size(op->type)) {
+                    expr = op;
+                    return;
+                }
+                if (ramp && stride && stride->value == 1) {
+                    int lanes = ramp->lanes;
+                    int vec_size = target.natural_vector_size(Int(8));
```

Comment:
I haven't missed this one. I'll work on this after lunch and then update the PR.